source/mongodb-handshake/handshake.rst

```diff
@@ -70,14 +70,18 @@ Connection handshake
 --------------------
```

Comment:
Bernie suggests adding a pseudocode section of the spec to mirror the paragraphs of scattered prose across the various specs. Perhaps something like this:
```
...

``hello`` and legacy hello commands issued after the initial connection handshake
MUST NOT contain handshake arguments. Any subsequent ``hello`` or legacy hello calls,
such as the ones for topology monitoring purposes, MUST NOT include this argument.

Example Implementation
~~~~~~~~~~~~~~~~~~~~~~

Consider the following pseudo-code for establishing a new connection:

.. code:: python

 conn = Connection()
 conn.connect()  # Connect via TCP / TLS
 if versioned_api_configured:
     cmd = {"hello": 1}
     conn.supports_op_msg = True  # Send the initial command via OP_MSG.
 else:
     cmd = {"legacy hello": 1, "helloOk": 1}
     conn.supports_op_msg = False  # Send the initial command via OP_QUERY.
 cmd["client"] = client_metadata
 if client_options.compressors:
     cmd["compression"] = client_options.compressors
 if client_options.load_balanced:
     cmd["loadBalanced"] = True
 creds = client_options.credentials
 if creds:
     # Negotiate auth mechanism and perform speculative auth. See Auth spec for details.
     if not creds.has_mechanism_configured():
         cmd["saslSupportedMechs"] = ...
     cmd["speculativeAuthenticate"] = ...
 
 reply = conn.send_command("admin", cmd)
 
 if reply["maxWireVersion"] >= 6:
     # Use OP_MSG for all future commands, including authentication.
     conn.supports_op_msg = True
 
 # Store the negotiated compressor, see OP_COMPRESSED spec.
 if reply.get("compression"):
     conn.compressor = reply["compression"][0]
 
 # Perform connection authentication. See Auth spec for details.
 negotiated_mechs = reply.get("saslSupportedMechs")
 speculative_auth = reply.get("speculativeAuthenticate")
 conn.authenticate(creds, negotiated_mechs, speculative_auth)

Hello Command
~~~~~~~~~~~~~

The initial handshake, as of MongoDB 3.4, supports a new argument, ``client``,
provided as a BSON object. This object has the following structure::
...
```

Reply:
Done.